security/pkg/pki/ra/k8s_ra.go

```diff
@@ -116,24 +116,17 @@ func (r *KubernetesRA) SignWithCertChain(csrPEM []byte, certOpts ca.CertOpts) ([
 		if err != nil {
 			return nil, fmt.Errorf("failed to find root cert from mesh config (%v)", err.Error())
 		}
-		if rootCertFromCertChain != nil && rootCertFromMeshConfig != nil {
```

Comment:
Some comments here would be super helpful. IMO, this logic is quite hard to read. Specifically, 

- IF there is a caCert specified in meshConfig, why we are we expecting that the CertChain of the signed cert from the external CA is signed by it
- What are the alternatives here - it appears that either the rootCert can be present in the certChain or not. What are the relevant scenarios where these could happen. Why does the MeshConfig root take precedence over the root in the certChain. 
- If we are extracting the "root" from the CertChain bytes, how come we are appending another "root" to it from MeshConfig. 
-

Reply:
the logic does look correct to me though, still wondering if more comments + refactoring would make this more readable